File: src/searchInsert/searchInsert.go
package searchInsert

//https://leetcode-cn.com/problems/search-insert-position/
func searchInsert(nums []int, target int) int {
	var L, R, mid = 0, len(nums) - 1, 0

	for L <= R {
		mid = L + ((R - L) >> 1)

		if nums[mid] < target {
			L = mid + 1
		} else {
			R = mid - 1
		}
	}

	return L
}

File: c/src/_0035/searchInsert.c
//
// Created by Administrator on 2022/10/10.
//

#include "searchInsert.h"

int searchInsert(int* nums, int numsSize, int target) {
    int l = 0;

    while (l < numsSize) {
        int mid  = l + (numsSize - l) / 2;
        if (nums[mid] > target) {
            numsSize = mid;
        } else {
            l = mid + 1;
        }
    }

    return l;
}